modules/modules.go

```diff
@@ -13,18 +13,20 @@ import (
 
 // Configuration holds module specific parameters
 type Configuration struct {
-	Name         string         `yaml:"name" json:"name"`
-	LdapGroups   []string       `yaml:"ldapgroups" json:"ldapgroups"`
-	UidMap       []umap         `yaml:"uidmap" json:"uidmap"`
-	Create       bool           `yaml:"create" json:"create"`
-	Delete       bool           `yaml:"delete" json:"delete"`
-	Notify       NotifyConf     `yaml:"notify" json:"notify"`
-	Credentials  interface{}    `yaml:"credentials" json:"credentials"`
-	Parameters   interface{}    `yaml:"parameters" json:"parameters"`
-	ApplyChanges bool           `yaml:"applychanges" json:"applychanges"`
-	NotifyUsers  bool           `yaml:"notifyusers" json:"notifyusers"`
-	LdapCli      mozldap.Client `yaml:"-" json:"-"`
-	Debug        bool           `yaml:"-" json:"-"`
+	Name          string         `yaml:"name" json:"name"`
+	LdapGroups    []string       `yaml:"ldapgroups" json:"ldapgroups"`
+	UidMap        []umap         `yaml:"uidmap" json:"uidmap"`
+	Create        bool           `yaml:"create" json:"create"`
+	Delete        bool           `yaml:"delete" json:"delete"`
+	Reset         bool           `yaml:"reset" json:"reset"`
+	ResetUsername string         `yaml:"resetusername" json:"resetusername"`
```

Comment:
The fact that we need to expose this variable through the module reinforces my opinion it should take the ldap uid, not the local uid, as argument.